### ISSUE TITLE
Issue when trying to create persistent object that already exists

### ISSUE THREAD

**Issue by HenrikRosenquistAndersson** [OP-TEE/optee_os#1933]
I downloaded the qemu_v8_stable.xml version yesterday and was working with the persistent objects when I ran into the following issue.

I can create a persistent object and then remove it. But if I try to create a object that already exists I get and error which is fine. But after that I'm not able to remove the object.

See sample code below. The first part works but the second part fails.
```c
#define TEE_ACCESS_REMOVE (TEE_DATA_FLAG_ACCESS_READ | TEE_DATA_FLAG_ACCESS_WRITE | TEE_DATA_FLAG_ACCESS_WRITE_META)
#define TEE_ACCESS_CREATE (TEE_DATA_FLAG_ACCESS_READ | TEE_DATA_FLAG_ACCESS_WRITE | TEE_DATA_FLAG_ACCESS_WRITE_META | TEE_DATA_FLAG_SHARE_READ)

static void testFile(void) {
  TEE_ObjectHandle hkey;
  TEE_Result res = TEE_SUCCESS;
  char name[] = "Test";
  char buf[32] = {0};

  EMSG("Start test- create file and delete it\n");

  res = TEE_CreatePersistentObject(TEE_STORAGE_PRIVATE_REE, name, 4, TEE_ACCESS_CREATE,
      NULL, (void*) buf, sizeof(buf), &hkey);
  if (TEE_SUCCESS == res)
    TEE_CloseObject(hkey);

  EMSG("TEE_CreatePersistentObject res:%x\n", res);

  res = TEE_OpenPersistentObject(TEE_STORAGE_PRIVATE_REE, name, 4, TEE_ACCESS_REMOVE, &hkey);
  EMSG("TEE_OpenPersistentObject res:%x\n", res);

  res = TEE_CloseAndDeletePersistentObject1(hkey);
  EMSG("TEE_CloseAndDeletePersistentObject1 res:%x\n", res);


  EMSG("Start test2- create file twice and delete it\n");
  res = TEE_CreatePersistentObject(TEE_STORAGE_PRIVATE_REE, name, 4, TEE_ACCESS_CREATE,
      NULL, (void*) buf, sizeof(buf), &hkey);
  if (TEE_SUCCESS == res)
    TEE_CloseObject(hkey);

  EMSG("TEE_CreatePersistentObject res:%x\n", res);

  res = TEE_CreatePersistentObject(TEE_STORAGE_PRIVATE_REE, name, 4, TEE_ACCESS_CREATE,
      NULL, (void*) buf, sizeof(buf), &hkey);
  if (TEE_SUCCESS == res)
    TEE_CloseObject(hkey);

  EMSG("TEE_CreatePersistentObject expect error res:%x\n", res);

  res = TEE_OpenPersistentObject(TEE_STORAGE_PRIVATE_REE, name, 4, TEE_ACCESS_REMOVE, &hkey);
  EMSG("TEE_OpenPersistentObject res:%x\n", res);

  res = TEE_CloseAndDeletePersistentObject1(hkey);
  EMSG("TEE_CloseAndDeletePersistentObject1 res:%x\n", res);
}
```

**Comment by jbech-linaro** [comment]
Hej @HenrikRosenquistAndersson ,

Sorry for a very late reply. This just caught my attention and I think we in the OP-TEE team have missed it. Did you get any further with this and more importantly do you know whether the bug is reproducible on latest? If it is, then I believe we need to dig into this.

// Regards
Joakim

**Comment by HenrikRosenquistAndersson** [comment]
Hello!
I just tried it on latest and couldn’t see the bug anymore. Since I did a workaround during autumn I haven’t
followed up the bug and don’t know when it disappeared.

Mit freundlichen Grüßen / Best regards

Henrik Andersson

Engineering Software, Lund, Sweden (RBSN/ESW11)
Tel. +46 46 270-6828 | Mobile +46 70 420-1628 | Henrik.Andersson@se.bosch.com<mailto:Henrik.Andersson@se.bosch.com>
